### PR TITLE
Onboarding checklist: change email_forwarding boolean to upgrade_email_forwarding task

### PR DESCRIPTION
**Pull request by robertf4**
#### Changes proposed in this Pull Request
This diff changes the way the email setup step for the onboarding checklist is sent from the server to the client.

Previously there was a single "email_setup" step that had an "email_forwarding" boolean property that could be toggled to prompt users with email forwarding to upgrade to G Suite.

Now there is an "email_setup" step as well as an "upgrade_email_forwarding" step. The "email_forwarding" boolean has been removed. This is being done in preparation for the next enhancement, which will add an additional "accept_gsuite_tos" step.

#### Testing instructions
Make sure you have D22540-code patched on to your sandbox or that it is merged.

Note that on some sites, particularly old ones, the checklist will not render by default. To get around this, set the `shouldRender` prop on the `WpcomChecklist` component to `true`.

To test, begin by make a site that satisfies domain and email settings for each of these cases, then go to `https://wordpress.com/checklist/<site_slug>`, and verify that you see the described task behavior.

| Domain                                       | Email                       | Expected Task Behavior                                                            |
---------------------------------|--------------------|-------------------------------------------------------------|
| A free *.wordpress.com domain | None                       | No email setup task is displayed                                            |
| Custom domain                          | None                       | Email setup task displays; shows option to purchase G Suite | 
| Custom domain                          | Email forwarding     | Email setup task displays; prompts an upgrade to G Suite     |
| Custom domain                          | G Suite                    | Email setup task displays; is marked as done                          |
| Custom domain                          | Custom MX record | Email setup task displays; is marked as done                          |

Also test that the task can be manually marked as done by pressing the empty circle on it when it is not completed.

### CODE REVIEW

File: client/my-sites/checklist/wpcom-checklist/wpcom-task-list.js [hunk 1]
```diff
@@ -54,8 +54,14 @@ export default class WpcomTaskList {
...
 
-		if ( config.isEnabled( 'onboarding-checklist/email-setup' ) && hasTask( 'email_setup' ) ) {
-			addTask( 'email_setup' );
+		if ( config.isEnabled( 'onboarding-checklist/email-setup' ) ) {
+			if ( hasTask( 'email_setup' ) ) {
+				addTask( 'email_setup' );
+			}
+
+			if ( hasTask( 'upgrade_email_forwarding' ) ) {
+				addTask( 'upgrade_email_forwarding' );
```
Comment (stephanethomas) [inline client/my-sites/checklist/wpcom-checklist/wpcom-task-list.js]: Shouldn't we name this task `email_forwarding_upgraded` to be consistent with other tasks?

**Comment by gravityrail** [comment]
On goldsounds.wordpress.com, which is on a Business plan but has no custom domain, I get an empty checklist. Is this expected?

empty-checklist

**Comment by gravityrail** [comment]
@robertf4 I get the same result on a site with a custom domain actually, kelleamiller.com

**Comment by robertf4** [comment]
> On goldsounds.wordpress.com, which is on a Business plan but has no custom domain, I get an empty checklist. Is this expected?

This is because that site is too old. To get around this, using React Dev Tools to set the `shouldRender` prop on the `WpcomChecklist` component to `true`.

**Comment by gravityrail** [comment]
Ok - I have the checklist showing now.

For `kelleamiller.com` the checklist API response shows:

```
email_setup: {completed: true, email_forwarding: false}
```

This is because it already has _external_ MX records.

Ok, this makes sense. And the setup step is rendering as "completed". 

However if I click on the email response _anyway_, it still takes me to http://calypso.localhost:3000/domains/manage/email/kelleamiller.com

We should disable this for sites where the task is completed.

**Comment by gravityrail** [comment]
I was able to reproduce correct behaviour for all sites _except_ "Custom domain with email forwarding", where I should expect "Email setup task displays; prompts an upgrade to G Suite"

I see the checklist displaying as completed, and no G Suite prompt.

checklist done

The checklist API response shows:

```
email_setup: {completed: true, email_forwarding: true}
```

File: client/my-sites/checklist/wpcom-checklist/component.jsx [hunk 2]
```diff
@@ -538,53 +539,56 @@ class WpcomChecklistComponent extends PureComponent {
...
+				} }
+				onDismiss={ this.handleTaskDismiss( task.id ) }
+			/>
+		);
+	};
 
-		const emailForwardingProps = task.email_forwarding
-			? emailForwardingUsed
-			: emailForwardingNotUsed;
+	renderEmailForwardingUpgradedToGSuiteTask = ( TaskComponent, baseProps, task ) => {
```
Comment (gravityrail) [inline client/my-sites/checklist/wpcom-checklist/component.jsx]: This title is misleading because of the `d` at the end of `Upgraded`

It should be `renderEmailForwardingUpgradeToGSuiteTask`

**Comment by gravityrail** [comment]
I deleted an earlier comment because I forgot to patch my sandbox 🤦‍♂️

File: client/my-sites/checklist/wpcom-checklist/component.jsx (hunk 2 above)
Comment (robertf4) [reply client/my-sites/checklist/wpcom-checklist/component.jsx]: It is supposed to have the `d`. All of the tasks are named using the past tense, e.g. `email_verified`, `site_created`, `address_picked`. So this task is called `email_forwarding_upgraded_to_gsuite`, and it is rendered by `renderEmailForwardingUpgradedToGSuiteTask()`.

**Comment by robertf4** [comment]
> We should disable this for sites where the task is completed.

I agree, it is disabled now.

File: client/my-sites/checklist/wpcom-checklist/component.jsx (hunk 2 above)
Comment (gravityrail) [reply client/my-sites/checklist/wpcom-checklist/component.jsx]: Oh I see! Thanks for the clarification.
